File: app/src/main/java/expressive/fan/SeekBarChangeListener.java
package expressive.fan;

import android.widget.SeekBar;

import expressive.fan.core.MainActivityController;

public class SeekBarChangeListener implements SeekBar.OnSeekBarChangeListener {

    private MainActivityController controller;

    public SeekBarChangeListener(MainActivityController controller) {
        this.controller = controller;
    }

    @Override
    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
        controller.progressChanged(progress, fromUser);
    }

    @Override
    public void onStartTrackingTouch(SeekBar seekBar) {

    }

    @Override
    public void onStopTrackingTouch(SeekBar seekBar) {

    }
}

File: app/src/main/java/expressive/fan/SeekBarChangeListener.kt
package expressive.fan

import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import expressive.fan.core.MainActivityController

class SeekBarChangeListener(private val controller: MainActivityController) : OnSeekBarChangeListener {
    override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
        controller.progressChanged(progress, fromUser)
    }

    override fun onStartTrackingTouch(seekBar: SeekBar?) {}
    override fun onStopTrackingTouch(seekBar: SeekBar?) {}

}